invokeai/frontend/web/src/features/nodes/util/graphBuilders/addHrfToGraph.ts

```diff
@@ -0,0 +1,210 @@
+import { RootState } from 'app/store/store';
+import { NonNullableGraph } from 'features/nodes/types/types';
+import {
+  DenoiseLatentsInvocation,
+  RescaleLatentsInvocation,
+  MetadataAccumulatorInvocation,
+  NoiseInvocation,
+  LatentsToImageInvocation,
+  Edge,
+} from 'services/api/types';
+import {
+  LATENTS_TO_IMAGE,
+  DENOISE_LATENTS,
+  NOISE,
+  MAIN_MODEL_LOADER,
+  METADATA_ACCUMULATOR,
+  LATENTS_TO_IMAGE_HRF,
+  DENOISE_LATENTS_HRF,
+  RESCALE_LATENTS,
+  NOISE_HRF,
+  VAE_LOADER,
+} from './constants';
+
+// Copy certain connections from previous DENOISE_LATENTS to new DENOISE_LATENTS_HRF.
+function copyConnectionsToDenoiseLatentsHrf(graph: NonNullableGraph): void {
+  const destinationFields = [
+    'vae',
+    'control',
+    'ip_adapter',
+    'metadata',
+    'unet',
+    'positive_conditioning',
+    'negative_conditioning',
+  ];
+  const newEdges: Edge[] = [];
+
+  // Loop through the existing edges connected to DENOISE_LATENTS
+  graph.edges.forEach((edge: Edge) => {
+    if (
+      edge.destination.node_id === DENOISE_LATENTS &&
+      destinationFields.includes(edge.destination.field)
+    ) {
+      // Add a similar connection to DENOISE_LATENTS_HRF
+      newEdges.push({
+        source: {
+          node_id: edge.source.node_id,
+          field: edge.source.field,
+        },
+        destination: {
+          node_id: DENOISE_LATENTS_HRF,
+          field: edge.destination.field,
+        },
+      });
+    }
+  });
+  graph.edges = graph.edges.concat(newEdges);
+}
+
+// Adds the high-res fix feature to the given graph.
+export const addHrfToGraph = (
+  state: RootState,
+  graph: NonNullableGraph
+): void => {
+  const { vae } = state.generation;
+  const isAutoVae = !vae;
```

Comment:
Maybe good to double check for `hrfToggled` at the start of this function

Reply:
added